assets/js/modules/analytics/components/common/EnableUniversalAnalytics.js

```diff
@@ -100,6 +106,31 @@ export default function EnableUniversalAnalytics( {
 		revertPropertyAndProfileIDs,
 	] );
 
+	const [ isLookingForMatch, setIsLookingForMatch ] = useState( false );
+
+	useEffect( () => {
+		const setMatchedProperty = async () => {
+			if ( ! propertyID ) {
+				setIsLookingForMatch( true );
+
+				const matchedProperty = await findMatchedProperty( accountID );
+				if ( matchedProperty?.id ) {
+					selectProperty( matchedProperty.id );
+				}
+
+				setIsLookingForMatch( false );
+			}
+		};
+
+		setMatchedProperty();
```

Comment:
While this works and looks alright, I think we usually use an anonymous async function in similar cases instead of declaring it as a function just to call it once. For example:

https://github.com/google/site-kit-wp/blob/02473040ee14ef681d48adc4956eff6bea7a3c1f/assets/js/modules/analytics-4/components/dashboard/ActivationBanner/index.js#L170-L177

Do you think we can/should do the same here?

Reply:
Yes, IIFE is a good choice here 👍